src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [Object.assign({}, state)];
+
+  for (const action of actions) {
+    const newState = Object.assign({}, stateHistory[stateHistory.length - 1]);
+
+    switch (action.type) {
```

Comment:
Use destruction for `action`
```suggestion
    const { type, ............ } = action;

    switch (type) {
```